### ISSUE TITLE
Awful Image Generation

### ISSUE THREAD

**Issue by shaun-ba** [NVlabs/Sana#93]
I mean what is this? I've followed all guides and reported various bugs for the past few days and this is the outcome?

![2024-12-17_08-02](https://github.com/user-attachments/assets/11e38bd9-27d3-4ff4-aa23-fa97f1ee0439)
![2024-12-17_08-03](https://github.com/user-attachments/assets/9baf770c-1ea0-49a4-90ff-d0f13c143a2c)
![2024-12-17_08-03_1](https://github.com/user-attachments/assets/8c754668-4c45-42a4-aa54-9d5601e78b1c)
![2024-12-17_08-04](https://github.com/user-attachments/assets/97900701-ad2a-451d-85f3-2e8bc31bd5bd)

**Comment by shaun-ba** [comment]
@lawrence-cj I cannot get anything to look realistic not just humans. See below for prompt "large oak tree". I thought your goals were to beat Flux for realism with 4x speed, but there isn't a single prompt that looks realistic to me, so I must be missing something?

![2024-12-17_08-13](https://github.com/user-attachments/assets/7b00197d-d59a-497d-8ccf-2174a6819676)

**Comment by shaun-ba** [comment]
Here is one from your homepage, prompt "a cyberpunk cat with a neon sign that says "SANA""

![2024-12-17_08-16](https://github.com/user-attachments/assets/8e3e3a59-4cf4-4229-971d-62f9cbd62b89)

**Comment by shaun-ba** [comment]
@lawrence-cj I've noticed that lowering CFG to 2.0-2.5 makes a massive difference with realism, under 2 then images don't look right, over 3 and they are as above kind of over exposed with lots of highlights on the edges of the detail etc.

Below is CFG 2.5, but even on this can you see on the ball of yarn the highlights on the top are blown out?

![2024-12-17_08-25](https://github.com/user-attachments/assets/7a5e800c-2673-44a8-8c5f-decdfe69c169)

**Comment by lawrence-cj** [comment]
Oh, if you want the content to be more realistic, then the prompt needs to be refined.
Here is a simple way to rewrite your prompt:
https://github.com/NVlabs/Sana/blob/bcad148693affa4f6a971516b57e75d2ee0b3221/app/app_sana.py#L64

BTW, we can't guarantee that Sana's smaller models will beat the FLUX just yet. Sana's larger model sizes and better VAEs are also in development and we have got pretty cool results. What we can guarantee is that we will keep improving and keep everything efficient. Maybe we still need time. 

At last, thank you for your time to play with Sana. We will keep working hard.

**Comment by lawrence-cj** [comment]
Besides, rewriting the prompt to a longer one will make the quality better in our experiment.

**Comment by shaun-ba** [comment]
@lawrence-cj I am used to writing very complex prompts with Flux with layering, background, middle ground, foreground etc. I was just testing as even in flux with a few words you get very good realistic results still. Using longer prompts there is generally more beneficial for finer control of the environment, subject, focus and such.

Using a few of your keywords above I don't get any better results. CFG does hugely effect the image though, I don't know if this is expected behavior or something odd going on with Comfy or schedulers etc?

Don't worry, I know this is a WIP and the speed is honestly outstanding, hopefully you can get "women lying on grass" on par with Flux!

CFG 4

![ComfyUI_Sana_00125_](https://github.com/user-attachments/assets/ec33b198-9fdc-47dc-b748-3b52a4ebc990)

CFG 2.5

![ComfyUI_Sana_00126_](https://github.com/user-attachments/assets/ca190a30-754b-49f5-84ee-7a0d2abaff8f)

**Comment by lawrence-cj** [comment]
> Using a few of your keywords above I don't get any better results. CFG does hugely affect the image though, I don't know if this is expected behavior or if something odd going on with Comfy or schedulers etc?

Thanks for your valuable testing results. I think "CFG will affect a lot" is a normal phenomenon. Higher CFG will make the image saturated and lower will make the image style more diverse and unstable. Is there no big difference between CFGs in FLUX? What kind of CFG is better during your testing?

Besides, our official Flow-DPM-Solver scheduler is not supported in ComfyUI for now. For the KSampler-Euler in ComfyUI, I didn't do a lot of experiments, I just made sure the workflow was working.

**Comment by bghira** [comment]
@lawrence-cj as flux is distilled, it has no CFG

**Comment by lawrence-cj** [comment]
> @lawrence-cj as flux is distilled, it has no CFG

Emmm, Flux-schnel has it.

**Comment by bghira** [comment]
that is fake guidance, not real cfg. it is distilled into that input argument and it behaves nothing like true cfg; you can use it up to 20.0 and beyond without ruining the image

**Comment by shaun-ba** [comment]
Sorry, as it stands even the 2k model is completely unusable for human images. If anyone wants to prove me wrong share a workflow but after hours of testing it cannot even deal with "woman sitting on a bench" without major flaws, not even in anatomy but the bench itself!

**Comment by bghira** [comment]
yeah we already saw the report you can stop complaining now. they already told you a new version will deal with it better. then he asked you questions about cfg that you didnt bother answering.

**Comment by shaun-ba** [comment]
> yeah we already saw the report you can stop complaining now. they already told you a new version will deal with it better. then he asked you questions about cfg that you didnt bother answering.
> 
> to be honest you were rude from the start.

I've likely spent more hours testing and reporting than most. I will continue to do this as I see fit, that isn't within your control.

There was no CFG question, just a statement that didn't help at all with realism.

**Comment by bghira** [comment]
the question you did not answer from lawrence was:

> Is there no big difference between CFGs in FLUX? What kind of CFG is better during your testing?

**Comment by bghira** [comment]
also you only ever opened one issue here and another one on the city96 repo with a rude title again, "Awful images": https://github.com/city96/ComfyUI_ExtraModels/issues/105

where's teh hours of testing and reporting?

**Comment by bghira** [comment]
i will continue to defend project maintainers from users that love to bully developers. it is a very productive use of time. you cannot stop or control me. does that sound familiar? now run along and stop ruining the issue tracker.
